src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,76 @@
+import { FC } from "react";
+import { IPaginationProps } from "./pagination.interface";
+import { StylesWrapper } from "./pagination.styles";
+import {
+  BsChevronLeft,
+  BsChevronRight,
+  BsChevronDoubleLeft,
+  BsChevronDoubleRight,
+} from "react-icons/bs";
+
+export const Pagination: FC<IPaginationProps> = ({
+  totalPage,
+  currentPage,
+  onPageClick,
+}) => {
+  if (!totalPage || totalPage == 1) {
+    return null;
+  }
+
+  const getClassName = (id: number) => {
+    if (id + 1 === currentPage) return "page-button active";
+    return "page-button";
+  };
+
+  const getPagesValidity = (id: number) =>
+    (currentPage - 1 <= 0 && id < 5) || // to check if current page is below 3
+    (currentPage + 2 > totalPage && totalPage - (id + 1) < 5) || // to check if current page is 1 less than the last page
+    Math.abs(id + 1 - currentPage) <= 2; // to only print 2 pages before and after current page
+
+  const getPages = (id: number) => {
```

Comment:
For functions that are meant to output JSX, you can use the prefix `render-` --> `renderPages()`

Reply:
🥔 donezies as well!